docs/release_notes/v1.16.2.md

```diff
@@ -228,3 +229,27 @@ When a workflow reminder is triggered, the workflow state is loaded from the act
 ### Solution
 
 Do not retry the reminder if the workflow state cannot be loaded, and instead log an error and exit the workflow execution.
+
+## Binding components now respect max-body-size annotation
+
+### Problem
+
+The `dapr.io/max-body-size` annotation only applied to incoming requests TO the Dapr sidecar, but not to responses FROM binding components (S3, Azure Blob, GCS, Minio, HTTP, etc.). This caused binding operations (GET, etc.) to fail when returning data larger than the default 4MB limit, even when users explicitly configured a higher limit via the annotation.
+
+### Impact
+
+Applications using binding components to fetch files or data larger than 4MB would fail with "Request body too large" errors, despite setting `dapr.io/max-body-size: "64Mi"` or higher. This affected all binding components that return data in their responses.
+
+### Root Cause
+
+The max-body-size limit was only enforced on the HTTP server's request handler for incoming requests. When binding components returned response data, the size check was bypassed, causing the default 4MB limit to be enforced regardless of the annotation value.
+
+### Solution
+
+Apply the max-body-size limit to binding component responses following the same pattern used for actor operations. The fix:
+- Wires through the `maxRequestBodySize` configuration from runtime to the binding processor
+- Adds response size validation after binding operations complete
+- Returns a clear error message when response exceeds the configured limit
+- Maintains backwards compatibility (zero limit bypasses the check)
+
+**Related Issue**: [components-contrib#4076](https://github.com/dapr/components-contrib/issues/4076)
```

Comment:
Please note I transferred the issue to dapr/dapr bc its a dapr bug. [see issue at this link.](https://github.com/dapr/dapr/issues/9177)